java-checks/src/main/java/org/sonar/java/checks/tests/NoTestInTestClassCheck.java

```diff
@@ -199,6 +201,28 @@ private static boolean checkRunWithType(Symbol.TypeSymbol value, String... runne
     return false;
   }
 
+  /**
+   * True if the symbol is annotated {@code @IncludeEngines("cucumber")}
+   * (with some approximation for automatic analysis).
+   */
+  private static boolean annotatedIncludeEnginesCucumber(Symbol.TypeSymbol symbol) {
+    for (var annotation: symbol.metadata().annotations()) {
+      if (annotation.symbol().type().fullyQualifiedName().endsWith("IncludeEngines")) {
+        // values are not available in automatic analysis, so assume "cucumber" is there
+        if (annotation.values().isEmpty()) {
+          return true;
+        }
+        // otherwise check the list
+        boolean containsCucumber = annotation.values().stream().anyMatch(annotationValue ->
+          Objects.deepEquals(annotationValue.value(), new Object[]{"cucumber"}));
```

Comment:
What happens here if we have a case where we have multiple values for the annotation? See this [example](https://github.com/yuzawa-san/googolplex-theater/blob/e3a3b2c7edb68646d6c6e5a4972577ab171ab427/src/test/java/com/jyuzawa/googolplex_theater/RunCucumber.java#L14)